Nef_S2/include/CGAL/Nef_S2/SM_point_locator.h

```diff
@@ -492,9 +492,7 @@ class SM_point_locator : public SM_decorator {
         } else {
           Sphere_segment first_half(p,p.antipode(),c);
           Sphere_segment second_part(p.antipode(),p,c);
-          if(!do_intersect_internally(ei->circle(), first_half, p_res)) {
-            (void)do_intersect_internally(ei->circle(), second_part, p_res);
-          }
+          CGAL_assertion( (! do_intersect_internally(ei->circle(), first_half, p_res) ) && do_intersect_internally(ei->circle(), second_part, p_res));
```

Comment:
That looks wrong. The assertion should be like `do_intersect_internally || do_intersect_internally`.

Reply:
yeah:

```c++
if(!a) throw;
if(b) throw;
```

is equivalent to:

```c++
if (!a || b) throw;
```